src/server/pkg/hashtree/db.go

```diff
@@ -1216,6 +1145,32 @@ func (mq *mergePQ) next() ([]*MergeNode, error) {
 	return ns, nil
 }
 
+func (mq *mergePQ) peek() ([]*MergeNode, error) {
+	if len(mq.q) == 0 || mq.size == 0 || mq.q[1] == nil {
+		return nil, errutil.ErrBreak
+	}
+	ns := []*MergeNode{mq.q[1].node}
```

Comment:
This could be a little cleaner by not putting the first node (1) in ns during initialization and then remove the check for cur != 1 before adding a node.

Reply:
Yup.